src/ngraph/runtime/gpu/gpu_kernel_emitters.cpp

```diff
@@ -45,6 +45,129 @@ void runtime::gpu::kernel::emit_memcpyDtD(codegen::CodeWriter& writer,
+                                                      const Shape& shape)
+{
+    std::vector<size_t> dimensions;
+    for (size_t i = shape.size(); i < 4; i++)
+    {
+        dimensions.push_back(1);
+    }
+    for (size_t i = 0; i < shape.size(); i++)
+    {
+        dimensions.push_back(shape[i]);
+    }
+
+    writer << "cudnnFilterDescriptor_t " << name << ";\n";
+    writer << "cudnnCreateFilterDescriptor(&" << name << ");\n";
+
+    if (dimensions.size() <= 4)
+    {
+        writer << "cudnnSetFilter4dDescriptor(" << name << ",\n";
+        writer << "                 /*dataType=*/" << data_type << ",\n";
+        writer << "                 /*format=*/" << format << ",\n";
+        writer << "                 /*dimension_size*/" << dimensions[0] << ",\n";
+        writer << "                 /*dimension_size*/" << dimensions[1] << ",\n";
+        writer << "                 /*dimension_size*/" << dimensions[2] << ",\n";
+        writer << "                 /*dimension_size*/" << dimensions[3] << ");\n";
+    }
+    else
+    {
+        writer << "const int " << name << "_axes[] = {" << join(dimensions) << "};\n";
+        writer << "cudnnSetFilterNdDescriptor(" << name << ",\n";
+        writer << "                 /*dataType=*/" << data_type << ",\n";
+        writer << "                 /*format=*/" << format << ",\n";
+        writer << "                 /*num_dimensions=*/" << dimensions.size() << ",\n";
+        writer << "                 /*dimensions*/" << name << "_axes);\n";
+    }
+}
+
+void runtime::gpu::kernel::emit_cudnnTensorDescriptor(codegen::CodeWriter& writer,
+                                                      const std::string& name,
+                                                      const std::string& format,
+                                                      const std::string& data_type,
+                                                      const Shape& shape)
+{
+    std::vector<size_t> dimensions;
+    for (size_t i = shape.size(); i < 4; i++)
+    {
+        dimensions.push_back(1);
+    }
+    for (size_t i = 0; i < shape.size(); i++)
+    {
+        dimensions.push_back(shape[i]);
+    }
+
+    writer << "cudnnTensorDescriptor_t " << name << ";\n";
+    writer << "cudnnCreateTensorDescriptor(&" << name << ");\n";
+
+    if (dimensions.size() <= 4)
+    {
+        writer << "cudnnSetTensor4dDescriptor(" << name << ",\n";
+        writer << "                 /*format=*/" << format << ",\n";
+        writer << "                 /*dataType=*/" << data_type << ",\n";
+        writer << "                 /*dimension_size*/" << dimensions[0] << ",\n";
+        writer << "                 /*dimension_size*/" << dimensions[1] << ",\n";
+        writer << "                 /*dimension_size*/" << dimensions[2] << ",\n";
+        writer << "                 /*dimension_size*/" << dimensions[3] << ");\n";
+    }
+    else
+    {
+        auto compute_strides = [](const std::vector<size_t>& dim) {
```

Comment:
nitpick: I wonder if @aprocter might have already written a helper for `compute_strides`?

Reply:
The function is `row_major_strides` in shape.hpp
I have been seeing this computation in other PRs. Can we clean this up?

Reply:
yes, let me change it. Thanks.